statistics/handle/update.go

```diff
@@ -891,17 +890,13 @@ func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) {
 				}
 				continue
 			}
-			if pruneMode == variable.StaticOnly || pruneMode == variable.StaticButPrepareDynamic {
-				for _, def := range pi.Definitions {
-					sql := "analyze table `" + db + "`.`" + tblInfo.Name.O + "`" + " partition `" + def.Name.O + "`"
-					statsTbl := h.GetPartitionStats(tblInfo, def.ID)
-					analyzed := h.autoAnalyzeTable(tblInfo, statsTbl, start, end, autoAnalyzeRatio, sql)
-					if analyzed {
-						return
-					}
-					continue
+			for _, def := range pi.Definitions {
+				sql := "analyze table `" + db + "`.`" + tblInfo.Name.O + "`" + " partition `" + def.Name.O + "`"
+				statsTbl := h.GetPartitionStats(tblInfo, def.ID)
+				analyzed := h.autoAnalyzeTable(tblInfo, statsTbl, start, end, autoAnalyzeRatio, sql)
+				if analyzed {
+					return
```

Comment:
We needn't analyze these partitions one by one since now `analyze partition-table` will analyze all partitions belonging to the table and merge them all to generate `global-stats`.

Reply:
If we `analyze partition-table`, auto analyze may be hard to be triggered since `partition-table` may be large and it may be rare for `ModifyCount` to reach the threshold. For a single partition, it may be more possible for `ModifyCount` to reach the threshold and trigger auto analyze. It is common that we only write to one partition in a period. In that case, analyzing one partition is easier to be triggered than analyzing the whole table.